server/src/main/java/org/apache/druid/server/QueryLaningStrategy.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import org.apache.druid.client.SegmentServerSelector;
+import org.apache.druid.query.QueryPlus;
+import org.apache.druid.server.scheduling.HiLoQueryLaningStrategy;
+import org.apache.druid.server.scheduling.NoQueryLaningStrategy;
+
+import java.util.Optional;
+import java.util.Set;
+
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "strategy", defaultImpl = NoQueryLaningStrategy.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "none", value = NoQueryLaningStrategy.class),
+    @JsonSubTypes.Type(name = "hilo", value = HiLoQueryLaningStrategy.class)
+})
+public interface QueryLaningStrategy
+{
+  /**
+   * Provide a map of lane names to the limit on the number of concurrent queries for that lane
+   * @param totalLimit
+   */
+  Object2IntMap<String> getLaneLimits(int totalLimit);
```

Comment:
is it expected that `sum(returned_map.values) < totalLimit` ? 
nit:  Also wonder why limits returned here  were not percentages but absolute counts given that each implementation would probably end up doing that calculation.

Reply:
>is it expected that sum(returned_map.values) < totalLimit ?

Right now I am not requiring this needs to be true since the limits are not guaranteed capacity, but rather maximums. It seemed more flexible to leave it to individual `QueryLaningStrategy` implementations to enforce that if they wish.

>nit: Also wonder why limits returned here were not percentages but absolute counts given that each implementation would probably end up doing that calculation.

Hmm, I think that is definitely worth considering, though @jihoonson was asking for absolute limits in this comment https://github.com/apache/druid/pull/9407#discussion_r389173425, so in the very least I will add a utility method to the `QueryLaningStrategy` interface for doing this conversion.

Reply:
Hehe, that happens with multiple reviewers. Yeah, having a utility method is equally good.